libraries/ESP8266WebServer/examples/FSBrowser/FSBrowser.ino

```diff
@@ -41,150 +77,180 @@ FS* filesystem = &LittleFS;
 
 const char* ssid = STASSID;
 const char* password = STAPSK;
-const char* host = "esp8266fs";
+const char* host = "fsbrowser";
 
 ESP8266WebServer server(80);
-//holds the current upload
-File fsUploadFile;
-
-//format bytes
-String formatBytes(size_t bytes) {
-  if (bytes < 1024) {
-    return String(bytes) + "B";
-  } else if (bytes < (1024 * 1024)) {
-    return String(bytes / 1024.0) + "KB";
-  } else if (bytes < (1024 * 1024 * 1024)) {
-    return String(bytes / 1024.0 / 1024.0) + "MB";
-  } else {
-    return String(bytes / 1024.0 / 1024.0 / 1024.0) + "GB";
-  }
+
+static bool fsOK;
+String unsupportedFiles = String();
+
+File uploadFile;
+
+
+////////////////////////////////
+// Utils to return HTTP codes, determine content-type and URLdecode params of GET method
+
+void returnOK() {
+  server.send(200, "text/plain", "");
+}
+
+void returnOKWithMsg(String msg) {
+  server.send(200, "text/plain", msg);
+}
+
+void returnNotFound(String msg) {
+  server.send(404, "text/plain", msg);
+}
+
+void returnFail(String msg) {
+  DBG_OUTPUT_PORT.println(msg);
+  server.send(500, "text/plain", msg + "\r\n");
 }
 
 String getContentType(String filename) {
-  if (server.hasArg("download")) {
-    return "application/octet-stream";
-  } else if (filename.endsWith(".htm")) {
+  if (filename.endsWith(".htm")) {
     return "text/html";
-  } else if (filename.endsWith(".html")) {
+  }
+  if (filename.endsWith(".html")) {
     return "text/html";
-  } else if (filename.endsWith(".css")) {
+  }
+  if (filename.endsWith(".css")) {
     return "text/css";
-  } else if (filename.endsWith(".js")) {
+  }
+  if (filename.endsWith(".js")) {
     return "application/javascript";
-  } else if (filename.endsWith(".png")) {
+  }
+  if (filename.endsWith(".png")) {
     return "image/png";
-  } else if (filename.endsWith(".gif")) {
+  }
+  if (filename.endsWith(".gif")) {
     return "image/gif";
-  } else if (filename.endsWith(".jpg")) {
+  }
+  if (filename.endsWith(".jpg")) {
+    return "image/jpeg";
+  }
+  if (filename.endsWith(".jpeg")) {
     return "image/jpeg";
-  } else if (filename.endsWith(".ico")) {
+  }
+  if (filename.endsWith(".ico")) {
     return "image/x-icon";
-  } else if (filename.endsWith(".xml")) {
+  }
+  if (filename.endsWith(".xml")) {
     return "text/xml";
-  } else if (filename.endsWith(".pdf")) {
+  }
+  if (filename.endsWith(".pdf")) {
     return "application/x-pdf";
-  } else if (filename.endsWith(".zip")) {
+  }
+  if (filename.endsWith(".zip")) {
     return "application/x-zip";
-  } else if (filename.endsWith(".gz")) {
+  }
+  if (filename.endsWith(".gz")) {
     return "application/x-gzip";
   }
   return "text/plain";
 }
```

Comment:
The octet-stream (download) case was moved up from getContentType() to the caller:
```
  if (server.hasArg("download")) {
    contentType = "application/octet-stream";
  } else {
    contentType = getContentType(path);
  }
```
because I found it weird for getContentType() to return a value based on the `server` global variable, and ignoring the parameter. 
Btw it now makes it compatible with the implementation you suggest from StaticRequestHandler :-)

Reply:
Anyway, using that StaticRequestHandler implementation makes perfect sense, but I admit I don't know how to call it... Tried this:
```
contentType = StaticRequestHandler::getContentType(path);
```
but I'm getting a `'StaticRequestHandler' has not been declared` error.
Do I need to include something specific ?
I guess my C++ limitations are showing up... :-(

Reply:
The goal is to refactor, not just replace the calls.
An example (off the top of my head) of what you could do is:
- pull the StaticRequestHandler<>::getContentType() function body out into a free standing function inside mime.h|cpp in the mime namespace.
- Replace the StaticRequestHandler::getContentType() body with a call to mime::getContentType() (i.e.: it becomes just a forwarder) to keep compatibility with current code
- #include "mime.h" in the sketch and use mime::getContetType() directly in the sketch if needed
- Add additional mime types to the mimetable if any are missing

This would unify code but still allow reuse by other users who need to getContentType() for whatever reason.

Reply:
Oh, I see.  OK, I didn't intend to jump out of the scope of this example.
I'll do my best, but what about making that refactoring a separate PR once this one is out ?
(By the way, comparing the code, it seems "jpeg" is missing from the StaticRequestHandler version)

Reply:
Alright, agreed. 
BTW, you can look me up in our [gitter channel](https://gitter.im/esp8266/Arduino), I could provide better guidance there, it would be much faster.